src/main/java/com/transportcompany/transportcompany/services/UserServiceImpl.java

```diff
@@ -1,22 +1,61 @@
 package com.transportcompany.transportcompany.services;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.transportcompany.transportcompany.models.dtos.UserDTO;
+import com.transportcompany.transportcompany.models.entities.User;
+import com.transportcompany.transportcompany.repositories.UserRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
 
+
+import java.util.ArrayList;
 import java.util.List;
 
+@Slf4j
+@Service
 public class UserServiceImpl implements UserService {
+
+
+//List <userDTO> userDTOList =new ArrayList<>();
+    private final UserRepository userRepository;
+    private final ObjectMapper objectMapper;
+
+    public UserServiceImpl(UserRepository userRepository, ObjectMapper objectMapper) {
+
+        this.userRepository = userRepository;
+        this.objectMapper = objectMapper;
+
+    }
+
     @Override
-    public UserDTO updateClient(UserDTO userDTO) {
-        return null;
+    public UserDTO createUser(UserDTO userDTO) {
+        User userToSave = objectMapper.convertValue(userDTO, User.class);
+        User userSaved = userRepository.save(userToSave);
+        return objectMapper.convertValue(userSaved, UserDTO.class);
     }
 
+
     @Override
-    public List<UserDTO> getAllUsers() {
-        return null;
+    public UserDTO updateUserById(long userId, UserDTO userDTO) {
+        User userFound = userRepository.findById(userId).orElseThrow(() -> new RuntimeException("User not found"));
+        userFound.setName(userDTO.getName());
+        userFound.setAddress(userDTO.getAddress());
+        userFound.setEmail(userDTO.getEmail());
+        User userSaved = userRepository.save(userFound);
+        return objectMapper.convertValue(userSaved, UserDTO.class);
     }
 
     @Override
     public List<UserDTO> getUsers() {
-        return null;
+        List<User> usersFound = userRepository.findAll();
+        List<UserDTO> usersFoundDTO = new ArrayList<>();
+        usersFound.forEach(user -> usersFoundDTO.add(objectMapper.convertValue(user, UserDTO.class)));
+        return usersFoundDTO;
+    }
+
+    public void deleteUserById(long id) {
+
```

Comment:
Please remove the empty row and consider checking if the user exists in the database before making the delete operation. We have to be sure that the user actually exists.